errors/error_test.go

```diff
@@ -198,3 +198,19 @@ func TestGetRootError(t *testing.T) {
 		}
 	}
 }
+
+func TestFieldsAdd(t *testing.T) {
+	fields := Fields{"key1": "val1", "key2": "val2"}
+	extFields := fields.Add("key3", "val3")
+	assert.Equal(t, "val3", extFields["key3"])
+	assert.Equal(t, nil, fields["key3"])
```

Comment:
Probably we don't have to check that we do not change the `fields` ancestor's state, not sure tho.